### PR TITLE
Move changes from #38681 and #39966 to compat 6.0

### PR DESCRIPTION
**Pull request by ajlende**
Moved compat/wordpress-5.9 changes that didn't make it into 5.9.x from #38681 and #39966 into compat/wordpress-6.0. 

See https://github.com/WordPress/gutenberg/pull/38681#issuecomment-1087431596 and https://github.com/WordPress/gutenberg/pull/39966#pullrequestreview-930214700

### CODE REVIEW

File: lib/load.php [hunk 1]
```diff
@@ -71,7 +71,7 @@ function gutenberg_is_experiment_enabled( $name ) {
 require __DIR__ . '/compat/wordpress-6.0/block-editor-settings.php';
 require __DIR__ . '/experimental/block-editor-settings-mobile.php';
 require __DIR__ . '/compat/wordpress-6.0/get-global-styles-and-settings.php';
-require __DIR__ . '/compat/wordpress-5.9/render-svg-filters.php';
```
Comment (ajlende) [inline lib/load.php:74]: @oandregal Seeing as the new function here completely replaces the old one, can the old file be removed?

Same question for the theme.json file that was updated.

Comment (oandregal) [reply lib/load.php:74]: Yeah, the old ones are no longer used.

**Review by oandregal — APPROVED** [review]

File: lib/load.php (hunk 1 above)
Comment (oandregal) [reply lib/load.php:74]: Pushed this myself and other minor updates to fast-track this PR.

**Comment by oandregal** [comment]
I've missed that there was a `WP_Theme_JSON_Resolver_Gutenberg` living at `lib/experimental` and a `WP_Theme_JSON_Resolver_6_0` living at `lib/compat/6.0` and that both had changes to the same method `get_theme_data`. 

I'm fixing it at https://github.com/WordPress/gutenberg/pull/40140

**Pull request merged**